sdk/core/core-http/lib/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,360 @@
+      response.parsedBody = atomResponseInJson;
+    }
+  } catch (e) {
+    errorBody = { code: "ResponseNotInAtomXMLFormat" };
+    return response;
+  }
+
+  // If received data is a non-valid HTTP response, the body is expected to contain error information
+  if (response.status < 200 || response.status >= 300) {
+    errorBody = atomResponseInJson;
+    if (errorBody == undefined) {
+      const HttpResponseCodes = Constants.HttpResponseCodes;
+      const statusCode = response.status;
+      const errorCode = isKnownResponseCode(statusCode)
+        ? HttpResponseCodes[statusCode]
+        : `UnrecognizedHttpResponseStatus: ${statusCode}`;
+      errorBody = {
+        code: errorCode
+      };
+    }
+  }
+
+  // Construct response with 'result' to be backward compatibile
+  const responseInCustomJson: { response: any; result: any } = {
+    response: null,
+    result: []
+  };
+
+  let isSuccessful = false;
+
+  if (errorBody == undefined) {
+    isSuccessful = true;
+    const result = parseAtomResult(atomResponseInJson);
+    if (result) {
+      if (Array.isArray(result)) {
+        result.forEach((entry: XMLResponseInJSON) => {
+          setName(entry, nameProperties);
+        });
+      } else {
+        setName(result, nameProperties);
+      }
+    }
+    responseInCustomJson.result = result;
+  } else {
+    // Transform the error info in response body to a normalized one
+    throw buildAtomError(errorBody, response);
+  }
+
+  responseInCustomJson.response = buildResponse(
+    isSuccessful,
+    response.parsedBody,
+    response.headers.rawHeaders(),
+    response.status
+  );
+
+  response.parsedBody = responseInCustomJson;
+  return response;
+}
+
+function isKnownResponseCode(
+  statusCode: number
+): statusCode is keyof typeof Constants.HttpResponseCodes {
+  return !!(Constants.HttpResponseCodes as { [statusCode: number]: string })[statusCode];
+}
+
+/**
+ * @ignore
+ * Extracts the applicable entity name(s) from the URL based on the known structure
+ * and instantiates the corresponding name properties to the deserialized response
+ *
+ * For instance, following is the URL structure for when creating a rule
+ * `<namespace-component>/<topic-name>/Subscriptions/<subscription-name>/Rules/<rule-name>`
+ *
+ * @param entry
+ * @param nameProperties
+ */
+function setName(entry: XMLResponseInJSON, nameProperties: any): any {
+  if (entry[Constants.ATOM_METADATA_MARKER]) {
+    const parsedUrl = new URL(entry[Constants.ATOM_METADATA_MARKER].id);
+
+    const parts = parsedUrl.pathname!.split("/");
+
+    for (let i = 0; i * 2 < parts.length - 1; i++) {
+      entry[nameProperties[i]] = parts[i * 2 + 1];
+    }
+  }
+}
+
+/**
+ * @ignore
+ * Utility to help construct the normalized error object based on given `errorBody`
+ * data and other data present in the received `response` object.
+ *
+ * @param errorBody
+ * @param response
+ */
+function buildAtomError(errorBody: any, response: HttpOperationResponse): AtomError {
```

Comment:
@bterlson, this function parses the response from the server and attempts to create an error object. Can you take a look at this and see if this is the standard we want to follow or is this too service specific?